tfx/examples/imdb/imdb_utils_native_keras.py

```diff
@@ -0,0 +1,215 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python source file include IMDB pipeline functions and necessary utils.
+
+The utilities in this file are used to build a model with native Keras.
+This module file will be used in Transform and generic Trainer.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import List, Text
+
+import absl
+import tensorflow as tf
+from tensorflow import keras
+import tensorflow_transform as tft
+
+from tfx.components.trainer.executor import TrainerFnArgs
+
+_FEATURE_KEY = 'text'
+_LABEL_KEY = "label"
+
+# There are 100 entries in the imdb_small dataset. ExampleGen splits the dataset
+# with a 2:1 train-eval ratio. Batch_size is an empirically sound
+# configuration.
+# To train on the entire imdb dataset, please refer to imdb_dataset_utils.py
+# and change the batch configuration accordingly.
+_DROPOUT_RATE = 0.2
+_EMBEDDING_UNITS = 64
+_EVAL_BATCH_SIZE = 5
+_HIDDEN_UNITS = 64
+_LEARNING_RATE = 1e-4
+_LSTM_UNITS = 64
+_VOCAB_SIZE = 8000
+_MAX_LEN = 400
+_TRAIN_BATCH_SIZE = 10
+
+def _transformed_name(key, is_input=False):
+  return key + ('_xf_input' if is_input else '_xf')
```

Comment:
just _xf? it's basically just a mark for transformed feature

https://github.com/tensorflow/tfx/blob/master/tfx/examples/iris/iris_utils_native_keras.py#L49

Reply:
for some reason, the keras embedding_layer is expecting _input at the end. So this is necessary.